mr/src/test/java/org/apache/iceberg/mr/TestInputFormatReaderDeletes.java

```diff
@@ -62,24 +67,20 @@ public static Object[][] parameters() {
     };
   }
 
-  @Before
+  @BeforeEach
   @Override
   public void writeTestDataFile() throws IOException {
     conf.set(CatalogUtil.ICEBERG_CATALOG_TYPE, Catalogs.LOCATION);
     super.writeTestDataFile();
   }
 
-  public TestInputFormatReaderDeletes(String inputFormat, FileFormat fileFormat) {
-    this.inputFormat = inputFormat;
-    this.fileFormat = fileFormat;
-  }
-
   @Override
   protected Table createTable(String name, Schema schema, PartitionSpec spec) throws IOException {
     Table table;
 
-    File location = temp.newFolder(inputFormat, fileFormat.name());
-    Assert.assertTrue(location.delete());
+    File location = temp.resolve(inputFormat).resolve(fileFormat.name()).toFile();
+    location.mkdirs();
```

Comment:
is this needed? other places in the codebase don't require this

Reply:
In other places, when we have a single temp file, we don't need it. 
But wherever we have nested directories, we do use it.
For example, in `TestPartitionValues`:

```
    String desc = "null_part";
    File parent = new File(temp.toFile(), desc);
    File location = new File(parent, "test");
    File dataFolder = new File(location, "data");
    assertThat(dataFolder.mkdirs()).as("mkdirs should succeed").isTrue();

```

Reply:
I can instead update the assert to `assertThat(location.mkdirs()).isTrue()`